### PR TITLE
rsClock: init at 0.1.0

### PR DESCRIPTION
**Pull request by valebes**
###### Motivation for this change
A simple tty clock written in Rust.
https://github.com/valebes/rsClock
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/applications/misc/rsclock/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{ stdenv, rustPlatform, fetchFromGitHub }:
+
+rustPlatform.buildRustPackage rec {
+  pname= "rsClock";
+  version= "0.1.0";
+
+  src = fetchFromGitHub {
+    owner = "valebes";
+    repo = pname;
+    rev ="49335356cd6d0544491cd97ede52e7e0ec027da2";
```
Comment (mmahut) [inline pkgs/applications/misc/rsclock/default.nix]: Do you mind either creating a github release of this software or move it to unstable as per https://nixos.org/nixpkgs/manual/#sec-package-naming

File: pkgs/applications/misc/rsclock/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    sha256 = "1fpidswkgpf9yr4vxqn38livz6r3z5i0lhg7ngj9f1ki4yqxn9zh";
+  };
+
+  checkPhase = "";
```
Comment (mmahut) [inline pkgs/applications/misc/rsclock/default.nix]: If you need to disable tests, please use ``doCheck = false;``.

Is there any reason they are not running?

File: pkgs/applications/misc/rsclock/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    rev ="49335356cd6d0544491cd97ede52e7e0ec027da2";
+    sha256 = "1fpidswkgpf9yr4vxqn38livz6r3z5i0lhg7ngj9f1ki4yqxn9zh";
+  };
+
+  checkPhase = "";    
+  cargoSha256 = "1m0lm8xh1qp0cbx870xy2m0bv047mb00vcwzq7r5gxqx8n61qm4n";
+
+  meta = with stdenv.lib; {
+    description = "A simple terminal clock written in Rust";
+    homepage = https://github.com/valebes/rsClock;
```
Comment (mmahut) [inline pkgs/applications/misc/rsclock/default.nix]: Use quotes please.

File: pkgs/applications/misc/rsclock/default.nix [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    license = stdenv.lib.licenses.mit;
+    maintainers = with maintainers; [valebes];
+    platforms = with platforms; linux ++ freebsd ++ openbsd ++ darwin;
```
Comment (mmahut) [inline pkgs/applications/misc/rsclock/default.nix]: Can this me ``platforms.all``?

**Comment by mmahut** [comment]
@GrahamcOfBorg build rsClock

**Comment by mmahut** [comment]
@GrahamcOfBorg build rsclock

File: pkgs/applications/misc/rsclock/default.nix (hunk 1 above)
Comment (valebes) [reply pkgs/applications/misc/rsclock/default.nix]: Ok

File: pkgs/applications/misc/rsclock/default.nix (hunk 2 above)
Comment (valebes) [reply pkgs/applications/misc/rsclock/default.nix]: This package don't have any tests.

File: pkgs/applications/misc/rsclock/default.nix (hunk 3 above)
Comment (valebes) [reply pkgs/applications/misc/rsclock/default.nix]: Fixed, thanks

File: pkgs/applications/misc/rsclock/default.nix (hunk 4 above)
Comment (valebes) [reply pkgs/applications/misc/rsclock/default.nix]: Ok

**Review by aanderse — CHANGES_REQUESTED** [review]
Thanks for choosing to contribute to nixpkgs! I've left a few comments I hope you'll find helpful.

Please squash/organize your commits such that you have a commit adding yourself to the maintainer list, followed by a commit adding the package.

File: pkgs/applications/misc/rsclock/default.nix [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    sha256 = "1fpidswkgpf9yr4vxqn38livz6r3z5i0lhg7ngj9f1ki4yqxn9zh";
+  };
+
+  doCheck = false;    
+  cargoSha256 = "1m0lm8xh1qp0cbx870xy2m0bv047mb00vcwzq7r5gxqx8n61qm4n";
+
+  meta = with stdenv.lib; {
+    description = "A simple terminal clock written in Rust";
+    homepage = "https://github.com/valebes/rsClock";
+    license = stdenv.lib.licenses.mit;
```
[the remark this answers is not in the document]
Comment (valebes) [reply pkgs/applications/misc/rsclock/default.nix]: Ok

File: pkgs/applications/misc/rsclock/default.nix [hunk 6]
```diff
@@ -0,0 +1,25 @@
+{ stdenv, rustPlatform, fetchFromGitHub }:
+
+rustPlatform.buildRustPackage rec {
+  pname= "rsClock";
+  version= "0.1.0";
```
[the remark this answers is not in the document]
Comment (valebes) [reply pkgs/applications/misc/rsclock/default.nix]: Ok, thanks :-)

File: maintainers/maintainer-list.nix [hunk 7]
```diff
@@ -6559,6 +6559,12 @@
     githubId = 1525767;
     name = "Vaibhav Sagar";
   };
+  valebes = {
```
[the remark this answers is not in the document]
Comment (valebes) [reply maintainers/maintainer-list.nix]: Fixed

File: pkgs/applications/misc/rsclock/default.nix [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+  version = "0.1.0";
+
+  src = fetchFromGitHub {
+    owner = "valebes";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1fpidswkgpf9yr4vxqn38livz6r3z5i0lhg7ngj9f1ki4yqxn9zh";
+  };
+
+  doCheck = false;
```
Comment (mmahut) [inline pkgs/applications/misc/rsclock/default.nix]: I'm really sorry, I got confused. ``doCheck`` is false by default so this can be removed.

Comment (valebes) [reply pkgs/applications/misc/rsclock/default.nix]: No problems! :-)

**Comment by valebes** [comment]
cc @aanderse 
Thanks for the tips!

**Review by aanderse — APPROVED** [review]
Looking good! Just squash the last commit into the previous and I think you're good for merge.

**Comment by alexarice** [comment]
I think you've squashed one too far here, I believe @aanderse wanted there to be 2 commits

**Comment by alexarice** [comment]
could try `git reflog`

**Comment by alexarice** [comment]
Definitely easier to revert to before when you made the rebase then redo it as the rebase has coauthored you in some commits

**Comment by valebes** [comment]
It was a nightmare.
I hope now will be all right.

**Comment by valebes** [comment]
Hooray!
Ready for merge.

**Comment by mmahut** [comment]
@GrahamcOfBorg build rsclock

**Review by alexarice — APPROVED** [review]
Commits look good to me now

**Review by aanderse — APPROVED** [review]
LGTM :+1:

**Comment by mmahut** [comment]
@valebes thank you so much for your contribution! Welcome to the community :)

**Comment by mmahut** [comment]
@GrahamcOfBorg build rsclock

**Comment by mmahut** [comment]
@valebes thank you for your contribution and welcome to the community! :)

**Comment by valebes** [comment]
@mmahut thank you for the supports! :-)

**Pull request merged**